File: vol1/ch4/LeetCode1047.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        deque<char> dq;
        for(int i=0;i<s.size();i++) {
            if (!dq.empty() && dq.back()==s[i])
                dq.pop_back();
            else
                dq.push_back(s[i]);
        }
        string ans="";
        while(!dq.empty()) {
            ans.push_back(dq.front());
            dq.pop_front();
        }
        return ans;
    }
};

File: vol1/ch4/LeetCode1047.py
from collections import deque
class Solution:
    def removeDuplicates(self, s: str) -> str:
        dq=deque()
        for i in range(0,len(s)):
            if dq and dq[-1]==s[i]:
                dq.pop()
            else:
                dq.append(s[i])
        return ''.join(dq)
